src/java/io/bazel/rulesscala/scalac/ScalaCInvoker.java

```diff
@@ -314,17 +314,42 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOEx
       });
     }
   }
-  private static void copyResources(Map<String, String> resources, Path dest) throws IOException {
+  private static void copyResources(
+      Map<String, String> resources,
+      String resourceStripPrefix,
+      Path dest) throws IOException {
     for(Entry<String, String> e : resources.entrySet()) {
       Path source = Paths.get(e.getKey());
       String dstr = e.getValue();
       if (dstr.charAt(0) == '/') dstr = dstr.substring(1);
+      // Check if we need to modify resource destination path
+      if (!"".equals(resourceStripPrefix)) {
+        dstr = getResourcePath(source, resourceStripPrefix);
+      }
```

Comment:
in the else branch is the only case where `e.getValue()` is used. Can we move it there to be clear:
```java
String dstr = null;
if (!"".equals(resourceStripPrefix)) {
  dstr = getResourcePath(source, resourceStripPrefix);
}
else {
  dstr = e.getValue();
}
// we can remove this line from getResourcePath now, right?
if (dstr.charAt(0) == '/') dstr = dstr.substring(1);
```